versioned_tree.go

```diff
@@ -154,34 +154,29 @@ func (tree *VersionedTree) GetVersioned(key []byte, version int64) (
 }
 
 // SaveVersion saves a new tree version to disk, based on the current state of
-// the tree. Multiple calls to SaveVersion with the same version are not allowed.
-func (tree *VersionedTree) SaveVersion(version int64) ([]byte, error) {
+// the tree. Returns the hash and new version number.
+func (tree *VersionedTree) SaveVersion() ([]byte, int64, error) {
+	version := tree.latestVersion + 1
+
 	if _, ok := tree.versions[version]; ok {
-		return nil, errors.Errorf("version %d was already saved", version)
+		return nil, version, errors.Errorf("version %d was already saved", version)
 	}
 	if tree.root == nil {
-		return nil, ErrNilRoot
-	}
-	if version == 0 {
-		return nil, errors.New("version must be greater than zero")
-	}
-	if version <= tree.latestVersion {
-		return nil, errors.Errorf("version must be greater than latest (%d <= %d)",
-			version, tree.latestVersion)
+		return nil, 0, ErrNilRoot
```

Comment:
Just ask for `tree.latestVersion = version` in this case with my rambling about multiple saves and matching block height.